projects/ui/src/hooks/chain/useTokenMap.ts

```diff
@@ -14,7 +15,9 @@ export default function useTokenMap<T extends Token>(
         // simply return the token. Otherwise we get the appropriate chain-
         // specific Token. This also dedupes tokens by address.
         const token = getChainToken(curr);
-        if (token) acc[token.address] = token;
+        // in the sdk, address of ETH is "". We need to use "eth" as key
+        const key = token instanceof Token && token.symbol === 'ETH' ? 'eth' : token.address;
+        if (token) acc[key] = token;
```

Comment:
This is more of a nitpick of mine than anything. Would this be better encapsulated in a function? Also, it would only evaluate the key if the token is truthy.

Ex,

```
// in the sdk, address of ETH is "". We need to use "eth" as key
const getTokenIndex = () => {
  if (token instanceof Token && token.symbol === 'ETH') {
    return 'eth';
  }

  return token.address;
}
...


...
if (token) acc[getTokenIndex(token)] = token;

```